datastore/transaction.go

```diff
@@ -110,17 +110,25 @@ func (c *Client) NewTransaction(ctx context.Context, opts ...TransactionOption)
 	return c.newTransaction(ctx, newTransactionSettings(opts))
 }
 
-func (c *Client) newTransaction(ctx context.Context, s *transactionSettings) (*Transaction, error) {
+func (c *Client) newTransaction(ctx context.Context, s *transactionSettings) (_ *Transaction, err error) {
 	req := &pb.BeginTransactionRequest{ProjectId: c.dataset}
 	if s.readOnly {
+		ctx = trace.StartSpan(ctx, "cloud.google.com/go/datastore.Transaction.ReadOnlyTransaction")
+		defer func() { trace.EndSpan(ctx, err) }()
+
 		req.TransactionOptions = &pb.TransactionOptions{
 			Mode: &pb.TransactionOptions_ReadOnly_{ReadOnly: &pb.TransactionOptions_ReadOnly{}},
 		}
-	} else if s.prevID != nil {
```

Comment:
Let's not modify the `else if` to `else`, this changes the actual logic of the function. Actually @BenWhitehead , would it be okay to just have a single "cloud.google.com/go/datastore.Client.newTransaction" for this function? It would prevent the confusing nested changes that we see here.

Reply:
We do need to keep all of the existing logic, so we'll need to keep the `else if`.

Due to the fact that read-write transactions can fail due to contention retries are often likely, so it would be useful to see multiple underlying RPCs attempted within the scope of a single application transaction request. (ReadOnly transactions aren't usually considered retryable because some of the parameters are time based and can easily expire across a retry attempt which could show the incorrect error to users).

Reply:
Okay, got it. So we can keep separate `ReadWrite` and `ReadOnly` in the two code branches here, but @AlisskaPie please revert the change you made to the logic that changed an `else if` into an `else`.